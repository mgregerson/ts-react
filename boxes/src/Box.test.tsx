import Box from "./Box";
import { render, fireEvent } from "@testing-library/react";

const mock = jest.fn();

describe("Box Component", function () {
  it("renders without crashing", function () {
    render(
      <Box id={1} width={5} height={5} backgroundColor="pink" remove={mock} />
    );
  });
  it("matches snapshot", function () {
    const { container } = render(
      <Box id={1} width={5} height={5} backgroundColor="pink" remove={mock} />
    );
    expect(container).toMatchSnapshot();
  });
  it("Displays a box on the page with appropriate styling", function () {
    const { container } = render(
      <Box id={1} width={5} height={5} backgroundColor="pink" remove={mock} />
    );
    const box = container.querySelector(".Box-box") as HTMLElement;
    const computedStyle = window.getComputedStyle(box);

    expect(computedStyle.backgroundColor).toEqual("pink");
    expect(computedStyle.width).toEqual("5em");
    expect(computedStyle.height).toEqual("5em");
  });
  it("Calls the remove function on click of the button", function () {
    const { container } = render(
      <Box id={1} width={5} height={5} backgroundColor="pink" remove={mock} />
    );
    const button = container.querySelector(".Box-removeBtn") as HTMLElement;
    fireEvent.click(button);
    expect(mock).toHaveBeenCalled();
  });
});
