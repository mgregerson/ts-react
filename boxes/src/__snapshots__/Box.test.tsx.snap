// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component matches snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="height: 5em; width: 5em; background-color: pink;"
    />
    <button
      class="Box-removeBtn"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;
